feat: Adding Python SDK for git step with Python3-9 runtime.

domderen opened the pull request:
This SDK is implementing the image contract(https://github.com/argoproj-labs/argo-dataflow/blob/main/docs/IMAGE_CONTRACT.md), by:
- Responding to /ready with a 204 status code once it is able to start serving requests,
- Accepting POST request to /messages, and responding with 201, 204 or 500 accordingly,
- Handles SIGTERM gracefully, giving server time to finish processing the request before terminating.

alexec commented:
Wow! Great stuff. When I wrote the Galang SDK, I updated runtimes/Galang-16 to use the SDK. Do you think we should do the same for this?

domderen commented:
Yeah, I'm doing it right now, give me a moment.

domderen commented:
Ok, so I added an example showing how to use this in a git step with a python runtime. I also added this:

```Dockerfile
RUN pip3 install git+https://github.com/argoproj-labs/argo-dataflow#subdirectory=sdks/python
```

To steps building the `python3-9` image. But I'm not sure where are python files inside this docker image, so I could modify them to use the new SDK. Can you point me in the right direction?

alexec commented on examples/106-git-pipeline.py:
```diff
@@ -6,13 +6,18 @@
          .git('main', 'https://github.com/argoproj-labs/argo-dataflow', 'main', 'examples/git', 'golang:1.16',
               env=[{'name': "GOCACHE", 'value': "/tmp/.gocache"}], command=["go", "run", "."])
-         .kafka('output-topic')
+         .kafka('go-output-topic')
```
Can we have a second separate Python example?

domderen replied:
As in `106-git-python-pipeline.py` that is a copy of the other one with a different runtime?

alexec commented on examples/106-git-pipeline.py:
```diff
@@ -6,13 +6,18 @@
+     .step(
+        (kafka('input-topic')
+         .git('main', 'https://github.com/argoproj-labs/argo-dataflow', 'main', 'examples/git-python', 'python:3.9', command=["./start.sh"])
+         .kafka('python-output-topic')
```
Examples are tested to make sure they can run. You’ll need to stick with input-topic and output-topic

alexec commented on Dockerfile:
```diff
@@ -89,6 +89,7 @@ ADD runtimes/python3-9 /workspace
 RUN chown -R 9653 /.cache /.local /workspace
 WORKDIR /workspace
 USER 9653:9653
+RUN pip3 install git+https://github.com/argoproj-labs/argo-dataflow#subdirectory=sdks/python
```
Should this go into requirements.txt?

domderen replied:
It should but I'm not sure where this requirements.txt lives. I couldn't find files that go into this image...

alexec commented:
I could merge this as the code needs to be on main to be built and pass test?

domderen commented on sdks/python/argo_dataflow_sdk/main.py:
```diff
@@ -0,0 +1,73 @@
+  def start(self, handler):
+    signal.signal(signal.SIGTERM, self.terminate)
+    MyServer.handler = handler
+    self.webServer = HTTPServer((hostName, serverPort), MyServer)
```
I'm wondering if we should use `ThreadingHTTPServer` rather than `HTTPServer`?

domderen commented:
Yeah, I think you can go for it.

alexec commented:
🚀
